driver/median_driver.hpp

```diff
@@ -0,0 +1,508 @@
+class MedianDriver : public Driver
+{
+public:
+    MedianDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputDesc);
+        miopenCreateTensorDescriptor(&inputGradDesc);
+        miopenCreateTensorDescriptor(&outputDesc);
+        miopenCreateTensorDescriptor(&outputGradDesc);
+        miopenCreateTensorDescriptor(&indicesDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    std::vector<int> ComputeStrides(std::vector<int> inputDim);
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyBackward() override;
+    int VerifyForward() override;
+
+    Tref GetTolerance();
+
+    ~MedianDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(inputGradDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+        miopenDestroyTensorDescriptor(outputGradDesc);
+        miopenDestroyTensorDescriptor(indicesDesc);
+    }
+
+private:
+    InputFlags inflags;
+    int forw;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t inputGradDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t outputGradDesc;
+    miopenTensorDescriptor_t indicesDesc;
+
+    std::unique_ptr<GPUMem> input_dev;
+    std::unique_ptr<GPUMem> input_grad_dev;
+    std::unique_ptr<GPUMem> output_dev;
+    std::unique_ptr<GPUMem> output_grad_dev;
+    std::unique_ptr<GPUMem> indices_dev;
+
+    std::vector<Tgpu> input;
+    std::vector<Tgpu> input_grad;
+    std::vector<Tgpu> output;
+    std::vector<Tgpu> output_grad;
+    std::vector<size_t> indices;
+
+    // Forward host
+    std::vector<Tref> output_host;
+    std::vector<size_t> indices_host;
+
+    // Backward host
+    std::vector<Tref> input_grad_host;
+
+    bool is_contiguous;
+    uint64_t dim;
+};
+
+// Equivalent tensor.transpose(0, -1).contiguous().transpose(0, -1)
+template <typename Tgpu, typename Tref>
+std::vector<int> MedianDriver<Tgpu, Tref>::ComputeStrides(std::vector<int> inputDim)
+{
+    if(!is_contiguous)
+    {
+        if(inputDim.size() == 1)
+            return std::vector<int>{2};
+
+        std::swap(inputDim.front(), inputDim.back());
+    }
+    std::vector<int> strides(inputDim.size());
+    strides.back() = 1;
+    for(int i = inputDim.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * inputDim[i + 1];
+    if(!is_contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+int MedianDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward Median (Default=1)", "int");
+    inflags.AddTensorFlag("input-dims",
```

Comment:
Next time, use smt like this for simplicity and consistency
```suggestion
    inflags.AddTensorFlag("input",
```